draft-wang-ppm-dap-taskprov.md

```diff
@@ -222,144 +225,212 @@ struct {
+> this to the Author?
+
+Finally, the Helper completes the aggregate initialize request as usual, deriving the VDAF
+verification key for the task as described in {{vdaf-verify-key}}.
 
 # Collector Behavior
 
-Collector should behave the same whether it supports `task-prov` extension or
-not. In particular, if leader does not recognize the task ID in the
-`CollectReq`, it will return HTTP status code 400 Bad Request with error type
-`unrecognizedTask` and an optional Retry-After header. Collector SHOULD retry
-with the same `CollectReq`, using interval in the Retry-After header if one
-exists.
+Upon receiving a `TaskConfig` from the Author, the Collector first decides
+whether to opt in to the task as described in {{provisioning-a-task}}. If the
+Collector opts out, it MUST not attempt to upload reports for the task.
 
-# Implementation and Operational Considerations
+Otherwise, once opted in, the Collector MAY begin to issue collect requests for
+the task. The task ID for each request MUST be derived from the `TaskConfig` as
+described in {{provisioning-a-task}}.
 
-> OPEN ISSUE: This mechanism brings added overhead in `Report` and `ReportShare`
-> since more duplicated data is passed around. Some optimisation can be done in
-> the core protocol to reduce this overhead.
+If the Leader responds to a collect request with an "unrecognizedTask" error,
+but the HTTP response includes a "Retry-After" header, the Collector SHOULD
+retry its collect request after waiting for the duration indicated by the
+header.
 
 # Security Considerations
 
-> NOTE(cjpatton) In this section we would describe any security goals we have
-> that go beyond the core DAP spec. We will also discuss if/how the extension
-> impacts the security of DAP itself.
+This document has the same security and privacy considerations as the core DAP
+specification. In particular, for privacy we consider the Author to be under
+control of the adversary. It is therefore incumbent on protocol participants to
+verify the privacy parameters of a task before opting in.
+
+In addition, the `task_prov` extension is designed to maintain robustness even
+when the Author misbehaves, or is merely misconfigured. In particular, if the
+Clients and Aggregators have an inconsistent view of the the task configuration,
+then aggregation of reports will fail. This is guaranteed by the binding of
+report metadata to encrypted input shares provided by HPKE encryption.
+
+> OPEN ISSUE: What if the Collector and Aggregators don't agree on the task
+> configuration? Consider adding
 
 A "task" now means the same task ID and same task configuration, if a malicious
 client changes the task ID or task configuration, its report will be aggregated
 in a different task, with other poison reports from the same malicious attack.
 The "good" reports will not be polluted.
 
+> CP: I don't understand the threat here. Is it that the client may lie about
```

Comment:
Yes, in the core protocol, client has nothing to lie about task provision, since task has been provisioned before client knows about it. I think the main point is malicious client who wants to send fake extension data will never be mixed with clients that have legitimate task configurations.

Reply:
I see.  IMO this doesn't need to be called out in security considerations. In any case, we should try to clean this up.

Reply:
If it doesn't add anything to security considerations, I'm fine deleting this paragraph

Reply:
Ack, I'll leave as-is, we can clean up later.

Reply:
https://github.com/wangshan/draft-wang-ppm-dap-taskprov/issues/13